### PR TITLE
Hide options that aren't supported in windows

### PR DESCRIPTION
**Pull request by karknu**

### CODE REVIEW

File: ouroboros-network/src/Ouroboros/Network/Socket.hs [hunk 1]
```diff
@@ -101,12 +102,15 @@ socketAsMuxBearer sd = do
       closeSocket = Socket.close sd
 
       sduSize :: IO Word16
+#if defined(mingw32_HOST_OS)
+      sduSize = return 13000 -- MaxSegment isn't supported on Windows
```
Comment (dcoutts) [inline ouroboros-network/src/Ouroboros/Network/Socket.hs]: How did we pick this value?

Comment (karknu) [reply ouroboros-network/src/Ouroboros/Network/Socket.hs]: I have a vague memory that you and Neil wanted 13000 but I'm unable to find any trace of it on slack.
Anyway it should be large but smaller than the initial window size (10*mss).

Comment (coot) [reply ouroboros-network/src/Ouroboros/Network/Socket.hs]: It would be good to place this in the comment.

**Pull request merged**